src/ProjectFileManager.cpp

```diff
@@ -1286,15 +1295,122 @@ class ImportProgress final
          }
       }
    }
-   
+
 private:
-   
+
    ImportFileHandle* mImportFileHandle {nullptr};
    std::unique_ptr<BasicUI::ProgressDialog> mProgressDialog;
 };
 
+namespace {
+void DoUseMirResultToConfigureProject(AudacityProject& project, double qpm)
+{
+   auto& projectTimeSignature = ProjectTimeSignature::Get(project);
+   projectTimeSignature.SetTempo(qpm);
+   // For now, our MIR only consists of filename parsing. If that succeeeds,
+   // most likely it is a 4/4 time signature. And if not, it still is something
+   // the user can adjust manually - until we get smarter.
+   projectTimeSignature.SetLowerTimeSignature(4);
+   projectTimeSignature.SetUpperTimeSignature(4);
+}
+
+enum class UserResponseToMirPrompt
+{
+   ManualYes,
+   PreferenceYes,
+   No,
+};
+
+UserResponseToMirPrompt UserWantsMirResultToConfigureProject(AudacityProject& project)
+{
+   const auto policy = UseMirResultToConfigureProject.Read();
+   if (policy == wxString("Ask"))
+   {
+      AudacityDontAskAgainMessageDialog m(
+         &GetProjectPanel(project), XO("Import"),
+         XO("Use detected music information to configure project?"));
+      const auto yes = m.ShowDialog();
+      if (m.IsChecked())
+         UseMirResultToConfigureProject.Write(
+            yes ? wxString("Yes") : wxString("No"));
+      return yes ? UserResponseToMirPrompt::ManualYes :
+                   UserResponseToMirPrompt::No;
+   }
+   else
+      return policy == wxString("Yes") ?
+                UserResponseToMirPrompt::PreferenceYes :
+                UserResponseToMirPrompt::No;
+}
+}
+
+void ReactOnMusicFileImport(
+   const std::string& fileName, const TrackHolders& newTracks,
+   AudacityProject& project)
+{
+   const auto waveTracks = newTracks[0]->Any<WaveTrack>();
```

Comment:
This assumes `newTracks` is not empty and no other subscript it used.

Maybe the argument should be `const Tracklist &`